src/main/java/com/wavefront/sdk/common/logging/MessageDedupingLogger.java

```diff
@@ -0,0 +1,68 @@
+package com.wavefront.sdk.common.logging;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+/**
+ * A logger that suppresses identical messages for a specified period of time.
+ *
+ * @author Han Zhang (zhanghan@vmware.com)
+ */
+public class MessageDedupingLogger extends DelegatingLogger {
+  private final long millisBetweenMessages;
+  private final Supplier<Long> clockMillis;
+  private final ConcurrentHashMap<String, AtomicLong> cache;
```

Comment:
this is unbounded in size and does not GC, we should just use a rate limiter and just acquire and log based on the type of log message instead of checking the exact string

Reply:
updated to use guava rate limiter and cache. cache and acquire based on a set of log message types specific to each class